templates/tfengine/components/cicd/main.tf

```diff
@@ -214,55 +210,57 @@ resource "google_service_account" "cloudbuild_scheduler_sa" {
 }
 
 resource "google_project_iam_member" "cloudbuild_scheduler_sa_project_iam" {
+  count = local.has_scheduled_jobs ? 1 : 0
+
   project = var.project_id
   role    = "roles/cloudbuild.builds.editor"
-  member  = "serviceAccount:${google_service_account.cloudbuild_scheduler_sa.email}"
+  member  = "serviceAccount:${google_service_account.cloudbuild_scheduler_sa[0].email}"
   depends_on = [
     google_project_service.services,
   ]
 }
-{{- end}}
 
 # Cloud Build - Cloud Build Service Account IAM permissions
-{{- if and $hasApplyJobs (get . "grant_automation_billing_user_role" true)}}
 
 # IAM permissions to allow Cloud Build Service Account use the billing account.
 resource "google_billing_account_iam_member" "binding" {
+  count = local.has_apply_jobs && var.grant_automation_billing_user_role ? 1 : 0
+
   billing_account_id = var.billing_account
   role               = "roles/billing.user"
   member             = local.cloudbuild_sa
   depends_on = [
     google_project_service.services,
   ]
 }
-{{- end}}
 
 # IAM permissions to allow Cloud Build SA to access state.
 resource "google_storage_bucket_iam_member" "cloudbuild_state_iam" {
   bucket = var.state_bucket
-  {{- if $hasApplyJobs}}
-  role   = "roles/storage.admin"
-  {{- else}}
-  role   = "roles/storage.objectViewer"
-  {{- end}}
+  role   = local.has_apply_jobs ? "roles/storage.admin" : "roles/storage.objectViewer"
   member = local.cloudbuild_sa
   depends_on = [
     google_project_service.services,
   ]
 }
 
-# Grant Cloud Build Service Account access to the {{.parent_type}}.
-resource "google_{{.parent_type}}_iam_member" "cloudbuild_sa_{{.parent_type}}_iam" {
-  {{- if $hasApplyJobs}}
-  for_each = toset(local.cloudbuild_sa_editor_roles)
-  {{- else}}
-  for_each = toset(local.cloudbuild_sa_viewer_roles)
-  {{- end}}
-  {{- if eq (get . "parent_type") "organization"}}
-  org_id   = {{.parent_id}}
-  {{- else}}
-  folder   = {{.parent_id}}
-  {{- end}}
+# Grant Cloud Build Service Account access to the organization.
+resource "google_organization_iam_member" "cloudbuild_sa_organization_iam" {
+  for_each = var.parent_type == "organization" ? local.has_apply_jobs ? toset(local.cloudbuild_sa_editor_roles) : toset(local.cloudbuild_sa_viewer_roles) : []
```

Comment:
Add () around `local.has_apply_jobs ? toset(local.cloudbuild_sa_editor_roles) : toset(local.cloudbuild_sa_viewer_roles)` so it's more clear that logic happens first?

Reply:
Done

I also tried to separate it into many lines but it doesn't work. The `terraform validate` throws an error.
If you know any possible solution to make it more readable let me know. I'm resolving this thread.